sklearn/manifold/tests/test_t_sne.py

```diff
@@ -879,6 +896,84 @@ def test_tsne_with_different_distance_metrics():
         assert_array_equal(X_transformed_tsne, X_transformed_tsne_precomputed)
 
 
+@pytest.mark.parametrize('method', ['exact', 'barnes_hut'])
+@ignore_warnings(category=FutureWarning)
+def test_tsne_with_legacy_euclidean_squaring(method):
+    """Make sure that TSNE preserves legacy Euclidean squaring behavior
+
+    Note: If 'legacy' is deprecated when the default for square_distance
+    is set to True, then this test can be removed, and 'euclidean' can
+    be included in `test_tsne_with_different_distance_metrics`."""
+    random_state = check_random_state(0)
+    n_components_original = 3
+    n_components_embedding = 2
+
+    # Used to create data with structure; this avoids unstable behavior in TSNE
+    X, _ = make_blobs(n_features=n_components_original,
+                      random_state=random_state)
+    X_precomputed = pairwise_distances(X, metric='euclidean', squared=True)
+
+    X_transformed_tsne = TSNE(
+        metric='euclidean', n_components=n_components_embedding,
+        random_state=0, square_distance="legacy",
+        method=method).fit_transform(X)
+    X_transformed_tsne_precomputed = TSNE(
+        metric='precomputed', n_components=n_components_embedding,
+        random_state=0, square_distance="legacy",
+        method=method).fit_transform(X_precomputed)
+
+    assert_array_equal(X_transformed_tsne, X_transformed_tsne_precomputed)
```

Comment:
assert_allclose is preferred when comparing float arrays

```suggestion
    assert_allclose(X_transformed_tsne, X_transformed_tsne_precomputed)
```

Reply:
Thank you, this makes sense. :)